.gitignore

```diff
@@ -1,10 +1,30 @@
-# Do not track objects or archives
+# Doe not track objects or archives
 *.o
 *.a
 
 # Do not track Makefiles, these are autogenerated
 *Makefile
 
+# Do not track various automake and libtool autogenerated files
+autom4te.cache
+*.lo
+*.la
+m4/libtool.m4
+m4/ltoptions.m4
+m4/ltsugar.m4
+m4/ltversion.m4
+m4/lt~obsolete.m4
+compile
+libtool
+ltmain.sh
+ar-lib
+.deps
+.libs
+.dirstamp
+
+# Do not track emacs backup files
```

Comment:
These editor-specific settings should be in a user's "global" `.gitignore` file, specified by the `core.excludesFile` variable in `$HOME/.gitconfig` (if you want these patterns to apply to all of the git repos you work on), or in `$GIT_DIR/info/exclude` (if you these patterns to apply only to this git repo) -- neither file should be tracked in the repo. (See the second bulleted list under "Description" in https://git-scm.com/docs/gitignore.) Delegating editor-specific `.gitignore` settings to these files eliminates the possibility of accumulating pattern cruft from multiple editors or IDEs (e.g., Sublime Text, Atom, TextMate, vim, nano, CLion, Eclipse, NetBeans, Visual Studio Code).